src/test/scala/vsys/blockchain/state/contract/vswap/RegisterVSwapContractDiffTest.scala

```diff
@@ -8,68 +8,84 @@ import org.scalatest.{Matchers, PropSpec}
+
+    /*** register liquidity token ***/
+    initLiquidityDataStack: Seq[DataEntry] <- initTokenDataStackGen(liquidityTotalSupply, liquidityUnity, "init")
+    description <- validDescStringGen
+    regLiquidityTokenContract <- registerTokenGen(master, tokenLiquidityContract, initLiquidityDataStack, description, fee + 10000000000L, ts+2)
+    liquidityTokenContractId = regLiquidityTokenContract.contractId
+    liquidityTokenId = tokenIdFromBytes(liquidityTokenContractId.bytes.arr, Ints.toByteArray(0)).explicitGet()
+
+    contract <- vSwapContract
     description <- validDescStringGen
-    create <- registerVSwapGen(master, contract, data, description, fee, ts + 1)
-  } yield (genesis, create)
 
-  property("register v-swap contract transaction doesn't break invariant") {
-    forAll(preconditionsAndRegContractTest) { case (genesis, reg: RegisterContractTransaction) =>
-      assertDiffAndState(Seq(TestBlock.create(Seq(genesis))), TestBlock.create(Seq(reg))) { (blockDiff, newState) =>
+    dataStack <- initVSwapContractDataStackGen(tokenAId.arr,tokenBId.arr,liquidityTokenId.arr, 100)
+    regContract <- registerVSwapGen(master, contract, dataStack, description, fee + 10000000000L, ts+3)
+
+    genesis <- genesisVSwapGen(master, ts)
+  } yield (genesis, regTokenAContract, regTokenBContract, regLiquidityTokenContract, regContract, fee + 10000000000L)
+
+
+  property("register VSwap contract function transactions doesn't break invariant") {
+    forAll(preconditionsAndVSwapWithTokenContractTest) { case (genesis, regA: RegisterContractTransaction, regB: RegisterContractTransaction, regLiquidity: RegisterContractTransaction, regVswap: RegisterContractTransaction, fee:Long) =>
+      assertDiffAndState(Seq(TestBlock.create(Seq(genesis, regA, regB, regLiquidity))), TestBlock.create(Seq(regVswap))) { (blockDiff, newState) =>
+
         val totalPortfolioDiff: Portfolio = Monoid.combineAll(blockDiff.txsDiff.portfolios.values)
-        totalPortfolioDiff.balance shouldBe -reg.transactionFee
-        totalPortfolioDiff.effectiveBalance shouldBe -reg.transactionFee
-        val master = reg.proofs.firstCurveProof.explicitGet().publicKey
-        val contractId = reg.contractId.bytes
+        totalPortfolioDiff.balance shouldBe -fee
+        totalPortfolioDiff.effectiveBalance shouldBe -fee
+        val master = regVswap.proofs.firstCurveProof.explicitGet().publicKey
+        val contractId = regVswap.contractId.bytes
         val makerKey = ByteStr(Bytes.concat(contractId.arr, Array(0.toByte)))
 
         val (_, masterTxs) = newState.accountTransactionIds(master, 2, 0)
         masterTxs.size shouldBe 2 // genesis, reg
-        newState.contractTokens(contractId) shouldBe 0
-        newState.contractContent(contractId) shouldEqual Some((2, reg.id, ContractVSwap.contract))
-
+        newState.contractContent(contractId) shouldEqual Some((2, regVswap.id, ContractVSwap.contract))
```

Comment:
just a simple comment about this test. compare with the previous one, it seems only added three token register contracts here. and the previous one just simply used systems token id directly. and the previous one also tested the build function. please re-write this test and make sure it covered more and clear.